traveltimepy/async_base_client.py

```diff
@@ -38,32 +48,25 @@ def __init__(
         max_rpm: int = 60,
         session: Optional[ClientSession] = None,
         use_ssl: bool = True,
+        split_large_requests: bool = True,
         _host: str = "api.traveltimeapp.com",
         _proto_host: str = "proto.api.traveltimeapp.com",
         _user_agent: str = f"Travel Time Python SDK {__version__}",
-        _split_size: int = 10,  # Splits requests to improve performance
     ):
-        self.app_id = app_id
-        self.api_key = api_key
-        self.timeout = timeout
-        self.retry_attempts = retry_attempts
-        self.max_rpm = max_rpm
+        super().__init__(
+            app_id=app_id,
+            api_key=api_key,
+            timeout=timeout,
+            retry_attempts=retry_attempts,
+            max_rpm=max_rpm,
+            use_ssl=use_ssl,
+            split_large_requests=split_large_requests,
+            _host=_host,
+            _proto_host=_proto_host,
+            _user_agent=_user_agent,
+        )
         self.session = session
-        self.use_ssl = use_ssl
-        self._host = _host
-        self._proto_host = _proto_host
-        self._user_agent = _user_agent
-
-        if _split_size <= max_rpm:
-            self._split_size = _split_size
-        else:
-            self._split_size = max_rpm
-
-        # TODO: Rework Async Limiter, right now it doesn't enforce limits correctly
-        self.async_limiter = AsyncLimiter(self.max_rpm // self._split_size)
-
-    def _build_url(self, endpoint: str) -> str:
-        return f"https://{self._host}/v4/{endpoint}"
+        self.async_limiter = AsyncLimiter(max_rate=self.max_rpm, time_period=60)
 
     async def _get_session(self):
         use_running_session = self.session and not self.session.closed
```

Comment:
I think we may keep the use_running_session logic for the session that we created, but not for the user provided session. Since we allow the user to provide his own session, this logic may hide the fact that his session got closed (or was closed from the beginning) and we created a new session for him, without notice. I'd rather have a fail-fast approach, were we raise an error if the user-provided session got unexpectedly closed.